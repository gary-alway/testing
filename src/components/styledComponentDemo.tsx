import styled from '@emotion/styled'
import React from 'react'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin: 1rem;
  padding: 2rem;
  border: 1px solid blue;
  background-color: yellow;
`

type Props = {
  className?: string
}

export const SomeStyledComponent: React.FC<Props> = ({ className }) => (
  <Container className={className}>
    this is styled! (and it can be custom styled because of the className prop)
  </Container>
)
