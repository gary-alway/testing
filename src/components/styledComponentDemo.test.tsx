import React from 'react'
import { render } from '@testing-library/react'
import { SomeStyledComponent } from './styledComponentDemo'

describe('styled components', () => {
  it('matches the snapshot', () => {
    const { container } = render(<SomeStyledComponent />)
    expect(container).toMatchSnapshot()
  })
})
