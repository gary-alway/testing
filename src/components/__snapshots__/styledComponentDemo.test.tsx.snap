// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled components matches the snapshot 1`] = `
<div>
  <div
    class="css-2rg88g"
  >
    this is styled! (and it can be custom styled because of the className prop)
  </div>
</div>
`;
